transport/http/inbound.go

```diff
@@ -182,16 +184,24 @@ func (i *Inbound) start() error {
 	return nil
 }
 
-// Stop the inbound, closing the listening socket.
+// Stop the inbound using Shutdown, but with a fixed timeout.
 func (i *Inbound) Stop() error {
-	return i.once.Stop(i.stop)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
We should thread this delay from an inbound option, then thread the inbound option through inbound config, with a default of 5 seconds.

Reply:
Agreed. I'm okay with landing this as-is and making that change before the next release, though.